src/main/java/build/buildfarm/worker/shard/Worker.java

```diff
@@ -515,13 +514,17 @@ private Server createServer(
           new WorkerProfileService(
               storage, inputFetchStage, executeActionStage, context, completeStage, backplane));
     } else {
-      PipelineStage casReplicationStage = new CasReplicationStage();
-      pipeline.add(casReplicationStage, 1);
+      createStorageOnlyPipeline(pipeline);
```

Comment:
Realistically it just doesn't need to have a pipeline running at all, and I'd prefer to have a dead pipeline (null or otherwise meaningless) that is interpreted by blockUntilShutdown as something that only hangs off of the server. That will obviate much of the need for the extra classes here and handling. Further reorganization should make the worker not have any knowledge of either server or pipeline, and just hang off of a critical service presentation layer to determine if it should shut down.

Reply:
for now, I removed the unnecessary pipeline stage.  We detect there are no pipeline stages and prevent shutdown.